backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/types/sort/impl/ElasticsearchDistanceSort.java

```diff
@@ -99,6 +103,13 @@ public void mode(SortMode mode) {
 
 		@Override
 		public SearchSort build() {
+			if ( missingFirst && SortOrder.ASC.equals( order ) ) {
```

Comment:
The order can be null, in which case the default will be used.

Please handle the `null` case and add a test for it.